polyply/tests/test_lib_files.py

```diff
@@ -0,0 +1,244 @@
+             '.pdb': compare_pdb}
+
+
+def _interaction_equal(interaction1, interaction2, inter_type):
+    """
+    Returns True if interaction1 == interaction2, ignoring rounding errors in
+    interaction parameters.
+    """
+    p1 = list(map(str, interaction1.parameters))
+    p2 = list(map(str, interaction2.parameters))
+    a1 = list(interaction1.atoms)
+    a2 = list(interaction2.atoms)
+    if inter_type in ["constraints", "bonds", "exclusions", "pairs"]:
+        a1.sort()
+        a2.sort()
+        return a1 == a2 and p1 == p2
+
+    elif inter_type in ["impropers", "dihedrals"]:
+         if frozenset([a1[0], a1[1], a1[2]]) == frozenset([a2[0], a2[1], a2[2]]) and a1[3] == a2[3] and p1 == p2:
+            return True
+         elif frozenset([a1[0], a1[1], a1[2]]) == frozenset([a2[3], a2[2], a2[1]]) and a1[3] == a2[0] and p1 == p2:
+            return True
+         elif frozenset([a1[1], a1[2], a1[3]]) == frozenset([a2[1], a2[2], a2[3]]) and a1[0] == a2[0] and p1 == p2:
+            return True
+
+    elif inter_type in ["angles"]:
+        return a1[1] == a2[1] and frozenset([a1[0], a1[2]]) == frozenset([a2[0], a2[2]]) and p1 == p2
+
+    return False
+
+@pytest.mark.parametrize("library, polymer", [
+     ['gromos2016H66', 'PP'],
+     ['gromos2016H66', 'C12E4'],
+     ['gromos2016H66', 'PE'],
+     ['gromos2016H66', 'PVA'],
+     ['gromos2016H66', 'PMA'],
+     ['gromos2016H66', 'PS'],
+   # ['martini3', 'PEO'],
+   # ['martini3', 'PS'],
+   # ['martini3', 'PEO_OHter'],
+   # ['martini3', 'P3HT'],
+    ['martini2', 'PEO'],
+    ['martini2', 'PS'],
+  # -> martini2 PEGylated lipids
+  # -> proteins?
+  # -> m2 polymers
+  #  ['gromos56A3', 'P3HT'],
+])
+def test_integration_protein(tmp_path, monkeypatch, library, polymer):
+    """
+    Runs tests on the library by executing the contents of the file `command` in
+    the folder libname/polymer, and tests whether the contents of the produced
+    files are the same as the reference files. The comparison of the files is
+    governed by `COMPARERS`.
+
+    Parameters
+    ----------
+    tmp_path
+    library: str
+    polymer: str
+    """
+    monkeypatch.chdir(tmp_path)
+
+    data_path = Path(PATTERN.format(path=INTEGRATION_DATA, library=library, polymer=polymer))
+
+    with open(str(data_path / 'command')) as cmd_file:
+        command = cmd_file.read().strip()
+
+    assert command  # Defensive
+    command = shlex.split(command)
+    result = [sys.executable]
+    for token in command:
+        if token.startswith('polyply'):  # Could be martinize2.py
+            result.append(str(POLYPLY))
+        elif token.startswith('.'):
+            result.append(str(data_path / token))
+        else:
+            result.append(token)
+
+    command = result
+
+    # read the citations that are expected
+    # citations = []
+    # with open(str(data_path/'citation')) as cite_file:
+    #     for line in cite_file:
+    #         citations.append(line.strip())
```

Comment:
Remove

Reply:
we already have the citations lined up and when they are there I can simply uncomment it. In this case I veto

Reply:
Accepted :)